apps/backend/charity/src/createCharity.ts

```
@@ -0,0 +1,100 @@
+  donated: number;
+  receiving: boolean;
+}
+
+export const createCharity = async (
+  req: AuthenticatedRequest,
+  res: Response,
+  db: IDatabase<object>,
+) => {
+  try {
+    const charity = req.body;
+    const { name, description, startDate, endDate, imageUrl, organizations } =
+      charity;
+
+    if (
+      !name ||
+      !description ||
+      !startDate ||
+      !endDate ||
+      !imageUrl ||
+      !organizations
+    ) {
+      console.error("missing parameter in request");
+      return res.status(400).json({ error: "missing parameter in request" });
+    }
+
+    for (const org of organizations) {
+      if (
+        !org.name ||
+        !org.logoUrl ||
+        org?.donated === null ||
+        org?.donated === undefined ||
+        org?.receiving === null ||
+        org?.receiving === undefined
+      ) {
+        console.error("missing parameter in request");
+        return res.status(400).json({ error: "missing parameter in request" });
+      }
+    }
+
+    const createdCharity = await db.oneOrNone(
+      `INSERT INTO charities (name, description, start_date, end_date, image_url) 
+      VALUES ($1, $2, $3, $4, $5) 
+      RETURNING charity_id, name, description, start_date, end_date, image_url`,
+      [name, description, startDate, endDate, imageUrl],
+    );
+
+    const queries = organizations.map((org: Organization) => {
+      return db.oneOrNone<Organization>(
+        `INSERT INTO organizations (name, logo_url, donated, receiving, charity_id)
+        VALUES ($1, $2, $3, $4, $5)
+        RETURNING name, logo_url, donated, receiving;`,
+        [
+          org.name,
+          org.logoUrl,
+          org.donated,
+          org.receiving,
+          createdCharity.charity_id,
+        ],
+      );
+    });
+
+    const results = await db.tx((t) => {
+      return t.batch(queries);
+    });
+
+    const createdOrganizations: Organization[] = results.filter(
+      (result) => result !== null,
+    ) as Organization[];
```

Would a null result mean an error has occured? In which case shouldn't we return an error (or such)?
 Why else would there be a null?